pkg/render/whisker/component.go

```diff
@@ -183,139 +146,21 @@ func (c *Component) whiskerService() *corev1.Service {
-	env := []corev1.EnvVar{
-		{Name: "LOG_LEVEL", Value: "INFO"},
-		{Name: "PORT", Value: "3002"},
-		{Name: "GOLDMANE_HOST", Value: "localhost:7443"},
-	}
-	volumeMounts := c.cfg.TrustedCertBundle.VolumeMounts(rmeta.OSTypeLinux)
-	if c.cfg.WhiskerBackendKeyPair != nil {
-		env = append(env,
-			corev1.EnvVar{
-				Name:  "TLS_KEY_PATH",
-				Value: c.cfg.WhiskerBackendKeyPair.VolumeMountKeyFilePath(),
-			},
-			corev1.EnvVar{
-				Name:  "TLS_CERT_PATH",
-				Value: c.cfg.WhiskerBackendKeyPair.VolumeMountCertificateFilePath(),
-			},
-		)
-		volumeMounts = append(volumeMounts, c.cfg.WhiskerBackendKeyPair.VolumeMount(c.SupportedOSType()))
-	}
-
 	return corev1.Container{
 		Name:            WhiskerBackendContainerName,
 		Image:           c.whiskerBackendImage,
 		ImagePullPolicy: render.ImagePullPolicy(),
-		Env:             env,
-		SecurityContext: securitycontext.NewNonRootContext(),
-		VolumeMounts:    volumeMounts,
-	}
-}
-
-func (c *Component) goldmaneContainer() corev1.Container {
-	var volumeMounts []corev1.VolumeMount
-
-	env := []corev1.EnvVar{
-		{Name: "LOG_LEVEL", Value: "INFO"},
-		{Name: "PORT", Value: "7443"},
-	}
-
-	if c.cfg.TrustedCertBundle != nil {
-		env = append(env,
-			corev1.EnvVar{
-				Name:  "CA_CERT_PATH",
-				Value: c.cfg.TrustedCertBundle.MountPath(),
-			},
-		)
-		volumeMounts = append(volumeMounts, c.cfg.TrustedCertBundle.VolumeMounts(c.SupportedOSType())...)
-	}
-
-	if c.cfg.GoldmaneServerKeyPair != nil {
-		env = append(env, corev1.EnvVar{
-			Name:  "SERVER_KEY_PATH",
-			Value: c.cfg.GoldmaneServerKeyPair.VolumeMountKeyFilePath(),
-		})
-		env = append(env, corev1.EnvVar{
-			Name:  "SERVER_CERT_PATH",
-			Value: c.cfg.GoldmaneServerKeyPair.VolumeMountCertificateFilePath(),
-		})
-
-		volumeMounts = append(volumeMounts, c.cfg.GoldmaneServerKeyPair.VolumeMount(c.SupportedOSType()))
-	}
-
-	if c.cfg.ManagementClusterConnection != nil {
-		env = append(env,
-			corev1.EnvVar{
-				Name:  "PUSH_URL",
-				Value: "https://localhost:8080/api/v1/flows/bulk",
-			},
-		)
-	}
-
-	return corev1.Container{
-		Name:            GoldmaneContainerName,
-		Image:           c.goldmaneImage,
-		ImagePullPolicy: render.ImagePullPolicy(),
-		Env:             env,
-		SecurityContext: securitycontext.NewNonRootContext(),
-		VolumeMounts:    volumeMounts,
-	}
-}
-
-func (c *Component) goldmaneService() *corev1.Service {
-	return &corev1.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      GoldmaneServiceName,
-			Namespace: WhiskerNamespace,
-		},
-		Spec: corev1.ServiceSpec{
-			Ports: []corev1.ServicePort{{Port: 7443}},
-			Selector: map[string]string{
-				"k8s-app": WhiskerDeploymentName,
-			},
+		Env: []corev1.EnvVar{
+			{Name: "LOG_LEVEL", Value: "INFO"},
+			{Name: "PORT", Value: "3002"},
+			{Name: "GOLDMANE_HOST", Value: "goldmane.calico-system.svc.cluster.local:7443"},
+			{Name: "TLS_CERT_PATH", Value: c.cfg.WhiskerBackendKeyPair.VolumeMountCertificateFilePath()},
```

Comment:
Strictly speaking this is introducing the possibility of a nil pointer dereference - I thought we usually had nil checks for this sort of thing.

Probably not the end of the world - we expect this to always be non-nil - but expectations can change over time so the safest code here would be to keep the nil check that we had before.

Reply:
I'm of the mind that this should panic if we're require this to be set. If there was an issue creating this then then controller needs to degrade.

Reply:
A panic is not going to cause the controller to degrade though.

I think it would be fine if you want to change the Component interface to return an `error` if it is missing required information, and bubble that up to the status manager. But that sounds like out of scope for this project, so I'd say we should follow the pattern we use elsewhere?

Reply:
But this is the pattern we follow, for things that must be configured. This is nothing other than a developer error (and panicing on developer errors is good), since it is created by the controller and if it's not then the user can't do anything about it.

Reply:
Yeah, I guess we are inconsistent in this. Sometimes we check nil, and sometimes we don't.

Panicking on developer errors is good, but it's much better to explicitly do that than to implicitly.

Like, maybe the constructor should have this:

```go
if cfg.WhickerBackendKeyPair == nil {
  logrus.Fatal("BUG: Missing required keypair %s")
}
```

Reply:
but fine if you just want to leave it as is, since we don't do this anywhere else.